solution/src/main/kotlin/com/cosmotech/solution/service/SolutionServiceImpl.kt

```diff
@@ -549,6 +603,42 @@ class SolutionServiceImpl(
         regexValidation = runTemplateParameterCreateRequest.regexValidation,
         options = runTemplateParameterCreateRequest.options)
   }
+
+  fun convertToRunTemplateParameterGroup(
+      runTemplateParameterGroupCreateRequest: RunTemplateParameterGroupCreateRequest
+  ): RunTemplateParameterGroup {
+    return RunTemplateParameterGroup(
+        id = runTemplateParameterGroupCreateRequest.id,
+        description = runTemplateParameterGroupCreateRequest.description,
+        labels = runTemplateParameterGroupCreateRequest.labels,
+        isTable = runTemplateParameterGroupCreateRequest.isTable,
+        options = runTemplateParameterGroupCreateRequest.options,
+        parentId = runTemplateParameterGroupCreateRequest.parentId,
+        parameters = runTemplateParameterGroupCreateRequest.parameters)
+  }
+
+  private fun checkParametersAndRunTemplateUnicity(
+      parameters: MutableList<RunTemplateParameterCreateRequest>?,
+      parameterGroups: MutableList<RunTemplateParameterGroupCreateRequest>?,
+      runTemplates: MutableList<RunTemplate>?
+  ) {
+    val hasDuplicateParameterId =
+        parameters?.groupBy { it.id.lowercase() }?.filterValues { it.size > 1 }?.isNotEmpty()
+            ?: false
+
+    val hasDuplicateParameterGroupId =
+        parameterGroups?.groupBy { it.id.lowercase() }?.filterValues { it.size > 1 }?.isNotEmpty()
+            ?: false
+
+    val hasDuplicateRunTemplateId =
+        runTemplates?.groupBy { it.id.lowercase() }?.filterValues { it.size > 1 }?.isNotEmpty()
+            ?: false
+
+    if (hasDuplicateParameterId || hasDuplicateParameterGroupId || hasDuplicateRunTemplateId) {
+      throw IllegalArgumentException(
+          "Several solution parameters or parameter groups or run templates have same id!")
```

Comment:
Should we be more specific and have a dedicated exception for each check? This would be more user-friendly than letting the users check all 3 lists for a duplicate.